torch/nested/_internal/nested_tensor.py

```diff
@@ -252,18 +276,14 @@ def __tensor_unflatten__(inner_tensors: Dict, meta, outer_size, outer_stride):
             metadata_cache["min_seqlen"] = min_seqlen_tensor
         if max_seqlen_tensor is not None:
             metadata_cache["max_seqlen"] = max_seqlen_tensor
-
         ragged_idx = meta["ragged_idx"]
 
-        # Note that we cannot simply check if is_fake(values) because
-        # during aot autograd, FunctionalTensors are not fake but hold
-        # symbolic sizes.
+        # Alternatively, we could make it the caller's responsibility to
+        # cache it. But this heuristic seems simple enough.
```

Comment:
I'm so sad to see this still here lol. I really want `tensor_unflatten()` to be straightforward and need to know nothing about fake tensors, tracing, etc.

I think the nested int should be cached during fake tensor construction for the strong invariant: a fake tensor post-creation is guaranteed to have any associated symbolic nested int set. I think this is done for one path during fake-ification but the `_empty_create_subclass()` path relies on `tensor_unflatten()` to set the nested int? Imo the responsibility to set it should be in fake-ification only and not in the scope of `tensor_unflatten()`.

Also, this logic will become irrelevant after the nested int revamp, so why not get closer to that ideal now :p

Reply:
> a fake tensor post-creation is guaranteed to have any associated symbolic nested int set

I had this earlier but removed it to make this PR smaller since it seemed like an orthogonal change. I like the idea of having this invariant earlier though, that is a great way to put it. I'll bring it back in an earlier PR.

Reply:
Opened a follow up PR https://github.com/pytorch/pytorch/pull/132414 to do this.